server/middleware/imageOptimizer.ts

```diff
@@ -1,15 +1,47 @@
 import * as express from 'express';
 import { expressSharp, HttpAdapter, S3Adapter } from 'express-sharp';
+import KeyvRedis from '@keyv/redis';
 import Keyv from 'keyv';
+import Redis from 'ioredis';
+
+class InvalidRedisURLError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = 'InvalidRedisURLError';
+  }
+}
 
 export default function imageOptimizer(): express.Router {
   let imageAdapter;
   let imageCache = null;
 
+  const redisUrl = process.env.REDIS_URL;
   if (process.env.REDIS_URL) {
-    imageCache = new Keyv(process.env.REDIS_URL, { namespace: 'image' });
+    let redis: Redis;
+    let keyvRedis: KeyvRedis;
+    if (redisUrl.startsWith('rediss://')) {
+      console.log('Attempting to connect to Redis with TLS..');
+      const options = { tls: { rejectUnauthorized: false } };
+      redis = new Redis(redisUrl, options);
+      // there is a bug in KeyvRedis which required me to also
+      // pass tls options there, otherwise it gets overrided.
```

Comment:
is this bug referenced somewhere (eg github issue)? if that's the case, it'd be good to mention it in the comment.

```suggestion
      // pass tls options there, otherwise it gets overridden.
```

Reply:
i can take a look to see if it is an issue that's been reported in KeyvRedis.. but i just found it and figured out the workaround.. if i can't find an issue i'll at least add some info here about the type of error i receive

Reply:
added a bit more detail to the comments, and then i also opened up an issue in `@keyv/redis`: https://github.com/jaredwray/keyv/issues/308